health-services/ExerciseSampleCompose/app/src/main/java/com/example/exercisesamplecompose/presentation/goals/ExerciseGoalsScreen.kt

```diff
@@ -0,0 +1,228 @@
+    var distanceValue by remember { mutableDoubleStateOf(0.0) }
+    var durationValue by remember { mutableDoubleStateOf(0.0) }
+
+    val columnState = rememberResponsiveColumnState(
+        contentPadding = ScalingLazyColumnDefaults.padding(
+            first = ItemType.Text,
+            last = ItemType.CompactChip,
+        )
+    )
+    ScreenScaffold(scrollState = columnState) {
+        ScalingLazyColumn(
+            modifier = Modifier.fillMaxSize(),
+            columnState = columnState,
+        ) {
+            if (!showDistanceRow) {
+                item {
+                    Chip(onClick = { showDistanceRow = !showDistanceRow },
+                        label = { Text(stringResource(R.string.add_distance)) })
+                }
+            }
+            if (showDistanceRow) {
+                item {
+                    DistanceGoalRow(distanceValue) { newValue ->
+                        distanceValue = newValue
+                    }
+                }
+            }
+            if (!showDurationRow) {
+                item {
+                    Chip(onClick = { showDurationRow = !showDurationRow },
+                        label = { Text(stringResource(R.string.add_duration)) })
+                }
+            }
+            if (showDurationRow) {
+                item {
+                    DurationGoalRow(durationValue) { newValue -> durationValue = newValue }
+                }
+            }
+            item {
+                Row(verticalAlignment = Alignment.CenterVertically) {
+                    CompactChip(label = { Text(stringResource(R.string.set_goal)) },
+                        // Set the goal and pass the values to the view model
+                        onClick = {
+                            setGoals(
+                                distanceValue * 1000, (durationValue * 60).toLong()
+                            ); onSet() // Close the screen
+                        })
+                }
+            }
+        }
+    }
+
+
+}
+
+
+@Composable
+fun DistanceGoalRow(distanceValue: Double, onDistanceValueChange: (Double) -> Unit) {
+    Column(horizontalAlignment = Alignment.CenterHorizontally) {
+        Text(
+            textAlign = TextAlign.Center,
+            text = stringResource(id = R.string.distance),
+            fontWeight = FontWeight.Bold
+        )
+        Spacer(modifier = Modifier.height(8.dp))
+        Row(verticalAlignment = Alignment.CenterVertically) {
+            CompactButton(onClick = {
+                val newValue = distanceValue - 0.5
+                onDistanceValueChange(newValue)
+            }) {
+                Icon(
+                    imageVector = Icons.Default.Remove,
+                    contentDescription = stringResource(R.string.minus)
+                )
+            }
+            Text(
+                textAlign = TextAlign.Center,
+                text = distanceValue.toString() + " " + stringResource(R.string.km)
```

Comment:
You could use "%d %s".format(distanceValue, stringResource(..."

or maybe call is distanceGoalTemplate, then : distanceGoalTemplate.format(val1, val2, ...) ?

Reply:
Done!